js/scripts.js

```diff
@@ -0,0 +1,67 @@
+// Testing if page width is mobile size.
+function testPageWidth() {
+  const breakPoint = 680,
+        treeContainer = document.getElementById("tree"),
+        trees = document.getElementsByClassName("hierplane");
+
+  // If desktop screen width:
+  if (window.innerWidth >= breakPoint) {
+    // If no trees exist
+    if (trees.length === 0 && pageId === "home") {
+      // Initialize Hierplane with data (see treeData.js)
+      hierplane.renderTree(introData, { target: "#tree" });
+    }
+    // If #tree container has mobile class
+    if (tree.classList.contains("mobile")) {
+      // Remove mobile class
+      treeContainer.classList.remove("mobile");
+    }
+  // Otherwise, if mobile screen width:
+  } else {
+    // Loop through .hierplane instances
+    while(trees[0]) {
+      // Remove each .hierplane
+      trees[0].parentNode.removeChild(trees[0]);
+    }
+    // Add mobile class
+    treeContainer.classList.add("mobile");
+  }
+}
+
+// Adding nav selection if link matches page
+function matchPage(e) {
+  if (e.getAttribute("data-page") === pageId) {
+    e.classList.add("nav__item--active");
+  }
+}
+
+// Set pageId and call matchPage
+const navLinks = Array.prototype.slice.call(document.querySelectorAll(".header__nav .nav__item[data-page]"));
+if (document.body.hasAttribute("data-page")) {
+  pageId = document.body.getAttribute("data-page");
+  navLinks.forEach(matchPage);
+}
+
+testPageWidth();
+// Initialize Syntax Highlighting
+hljs.initHighlightingOnLoad();
```

Comment:
How big is the highlight file? If it's big, let's omit it and just go without syntax highlighting.

Reply:
It's 12k. There's another one called Prism that claims to be 2k gzipped: http://prismjs.com/

I just went with highlight.js since that's what we used on AllenNLP.

Reply:
2k would be better, is there any downside?

Reply:
Will take a stab at using Prism instead in next pass.

Reply:
After downloading Prism, I discovered its core JS is 11kb without gzip. Highlight JS is 12kb. I'm thinking less than 1k of savings when gzipped is probably not worth the time it would take to reimplement this feature with another library.